src/System.Net.Sockets/src/System/Net/Sockets/_MultipleConnectAsync.cs

```diff
@@ -31,6 +39,10 @@ private enum State
 
         private object _lockObject = new object();
 
+        protected abstract bool RequiresUserConnectAttempt { get; }
```

Comment:
Instead of changing the global code to accept various OS behaviors, we should instead push the behavior down within PAL and conditionally compile based on what the platform supports.